modular_zubbers/code/game/gamemodes/fleshmind/fleshmind_environment_objects.dm

```diff
@@ -0,0 +1,152 @@
+	var/object_attack_damage = 150
+	/// Are we active?
+	var/active = FALSE
+	/// Are we a vent burrow?
+	var/vent_burrow = FALSE
+	/// ZOnes we passively replace
+	var/static/list/replacement_zones = list(
+		BODY_ZONE_L_ARM = /obj/item/bodypart/arm/left/robot,
+		BODY_ZONE_L_LEG = /obj/item/bodypart/leg/left/robot,
+		BODY_ZONE_R_ARM = /obj/item/bodypart/arm/right/robot,
+		BODY_ZONE_R_LEG = /obj/item/bodypart/leg/right/robot,
+	)
+
+/obj/structure/fleshmind/wireweed/Initialize(mapload, starting_alpha = 255, datum/fleshmind_controller/incoming_controller)
+	. = ..()
+	alpha = starting_alpha
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+	our_controller = incoming_controller
+
+/obj/structure/fleshmind/wireweed/wirecutter_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(tool.use_tool(src, user, WIREWEED_WIRECUTTER_KILL_TIME, volume = 50))
+		if(QDELETED(src))
+			return
+		qdel(src)
+
+/obj/structure/fleshmind/wireweed/update_icon(updates)
+	. = ..()
+	if((updates & UPDATE_SMOOTHING) && (smoothing_flags & (SMOOTH_BITMASK)))
+		if(!vent_burrow)
+			QUEUE_SMOOTH(src)
+		QUEUE_SMOOTH_NEIGHBORS(src)
+
+/obj/structure/fleshmind/wireweed/update_icon_state()
+	. = ..()
+	if(vent_burrow)
+		icon_state = "vent_burrow"
+
+/obj/structure/fleshmind/wireweed/emp_act(severity)
+	. = ..()
+	switch(severity)
+		if(EMP_LIGHT)
+			take_damage(20)
+		if(EMP_HEAVY)
+			take_damage(40)
+
+
+/obj/structure/fleshmind/wireweed/update_overlays()
+	. = ..()
+	if(active)
+		. += "active"
+	for(var/wall_dir in GLOB.cardinals)
+		var/turf/new_turf = get_step(src, wall_dir)
+		if(new_turf && new_turf.density) // Assume we are a wall!
+			var/image/new_wall_overlay = image(icon, icon_state = "wall_hug", dir = wall_dir)
+			switch(wall_dir) //offset to make it be on the wall rather than on the floor
+				if(NORTH)
+					new_wall_overlay.pixel_y = 32
+				if(SOUTH)
+					new_wall_overlay.pixel_y = -32
+				if(EAST)
+					new_wall_overlay.pixel_x = 32
+				if(WEST)
+					new_wall_overlay.pixel_x = -32
+			. += new_wall_overlay
+
+/obj/structure/fleshmind/wireweed/proc/visual_finished()
+	SIGNAL_HANDLER
+	alpha = 255
+
+/obj/structure/fleshmind/wireweed/proc/on_entered(datum/source, atom/movable/moving_atom)
+	SIGNAL_HANDLER
+	if(!isliving(moving_atom))
+		return
+	if(istype(moving_atom, /mob/living/basic/fleshmind/tyrant)) // Don't heal this thing
+		return
+	var/mob/living/entered_mob = moving_atom
+	if(!faction_check(entered_mob.faction, faction_types))
+		return
+	if(prob(WIREWEED_HEAL_CHANCE))
+		entered_mob.heal_overall_damage(WIREWEED_HEAL_AMOUNT, WIREWEED_HEAL_AMOUNT)
+		to_chat(entered_mob, span_green("[src] heals you as you cross over it!"))
+	if(ishuman(entered_mob) && prob(WIREWEED_REPLACE_BODYPART_CHANCE))
+		var/mob/living/carbon/human/human_mob = moving_atom
+		for(var/zone as anything in replacement_zones)
```

Comment:
does this consider unremovable limbs, such as taurs? if not, theres a limb flag for this

Reply:
oh wait nvm im dumb